sdk/storage/blobs/inc/az_storage_blobs.h

```diff
@@ -33,48 +47,76 @@ typedef struct
   struct
   {
     // buffer to copy customer url. Then it stays immutable
-    uint8_t url_buffer[AZ_HTTP_REQUEST_URL_BUF_SIZE];
-    // this url will point to url_buffer
-    az_span uri;
+    uint8_t endpoint_buffer[AZ_HTTP_REQUEST_URL_BUF_SIZE];
+    // this url will point to endpoint_buffer
+    az_span endpoint;
     _az_http_pipeline pipeline;
     az_storage_blobs_blob_client_options options;
     _az_credential* credential;
   } _internal;
 } az_storage_blobs_blob_client;
 
+/**
+ * @brief Initialize a client with default options.
+ *
+ * @param client The blob client instance to initialize.
+ * @param endpoint A url to a blob storage account.
+ * @param credential credential object for authentication
+ *         #AZ_CREDENTIAL_ANONYMOUS should be used for SAS
+ * @param options client options
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ */
 AZ_NODISCARD az_result az_storage_blobs_blob_client_init(
     az_storage_blobs_blob_client* client,
-    az_span uri,
+    az_span endpoint,
     void* credential,
     az_storage_blobs_blob_client_options* options);
 
 typedef struct
 {
-  az_span option;
+  struct
+  {
+    az_span option;
```

Comment:
nit: In case you missed it, rename to `unused`. And maybe make it a char/bool to keep the struct small?
```suggestion
    az_span unused;
```